pkg/cmd/util/clientcmd/factory.go

```diff
@@ -234,17 +238,19 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 	}
 	kLogsForObjectFunc := w.Factory.LogsForObject
 	w.LogsForObject = func(object, options runtime.Object) (*kclient.Request, error) {
-		oc, _, err := w.Clients()
```

Comment:
Why not use a similar check for `OriginKind`?  I'd be willing accept the reflective cost.  `api.Scheme.ObjectVersionAndKind` I think.